### ISSUE TITLE
Multiple document addition or update tasks failing due to 'max_fields_limit_exceeded'

### ISSUE THREAD

**Issue by qdequele** [meilisearch/meilisearch#3833]
A user, is encountering an issue where all the tasks fail. The error message received is: "A document cannot contain more than 65,535 fields." This is happening despite the fact that the documents being uploaded do not contain more than 60 fields.

**Description**

The user has a large number of tasks (18M) in the queue, sending documents one by one. This error is received for each task. However, upon inspecting the index stats, it seems that only 60 fields are used across 63k documents.

This error is believed to be due to multiple schema updates (fields names) that may have taken place throughout the project history.

Here is an example of a failed task:
```json
{
    "uid": 16760922,
    "indexUid": "index",
    "status": "failed",
    "type": "documentAdditionOrUpdate",
    "canceledBy": null,
    "details": {
        "receivedDocuments": 1,
        "indexedDocuments": 0
    },
    "error": {
        "message": "A document cannot contain more than 65,535 fields.",
        "code": "max_fields_limit_exceeded",
        "type": "invalid_request",
        "link": "https://docs.meilisearch.com/errors#max_fields_limit_exceeded"
    },
    "duration": "PT0.058707778S",
    "enqueuedAt": "2023-06-13T00:59:51.631900659Z",
    "startedAt": "2023-06-13T00:59:51.642606814Z",
    "finishedAt": "2023-06-13T00:59:51.701314592Z"
}
```
We need to find a solution to this problem as it is affecting the user's ability to fully utilize Meilisearch.

**Meilisearch version:**
v1.2.0

**Additional context**
Cloud - Pro users

**Comment by qdequele** [comment]
So, in fact, the issue is reopening. 

For context, the customer has a B2C app, a Twitter-like app. They are using Firestore and Firestore sends the entire documents directly to Meilisearch. However, in the documents sent by Firestore to Meilisearch, they have a field containing all the people that liked a post. Unfortunately, this list is not an array of user IDs but an object of user IDs, and they are the key!

It looks like this:
```json
{
    "likes": {
        "111": true, 
        "222": false,
        "333":  true
    }
}
```
Instead of this:
```json
{
    "likes": ["111", "222", "333"]
}
```

**Comment by Kerollmops** [comment]
> Today, this is a deal-breaker. I'll try to find an alternative solution, but I'm not sure if it will work.

Why can't the user switch to something that looks like this instead? Could you explain the exact use-case to me, please? 

```
{
    "likes_true": ["111", "333"],
    "dont_false": ["222"]
}
```

> @Kerollmops, you told me about a possible fix by removing the unused fields from the list of fields used at some point. How long would it take to fix?

The fix I told you about will only work if there are fewer than 65,536 fields over all of the documents in the index. Nested field counts, i.e., `likes.111`, `likes.222`. We can potentially release a prototype with a version of Meilisearch using four bytes integers instead of 2 bytes integers. Unfortunately, we didn't plan to change that anytime soon. It will take some time.

Can't we find a better solution by understanding the exact use case, before diving into this?

**Comment by qdequele** [comment]
Thanks for responding. I’ll continue the discussion with the user to understand why he can’t change the format of the documents they send to Meilisearch. I’ll keep you updated. 

Is there a world where having a setting like “ignoredField” would be a good solution?

**Comment by qdequele** [comment]
I don't know. The only thing I want is not to create an internal ID for the fields that are not used for indexation. How complicated would it be to add a patch to create an internal ID only for fields that are searchable, filterable, or sortable?

**Comment by Kerollmops** [comment]
> The only thing I want is not to create an internal ID for the fields that are not used for indexation.

You are absolutely right. That is a thing that we should be able to do in future versions of the engine.

> How complicated would it be to add a patch to create an internal ID only for fields that are searchable, filterable, or sortable?

I haven't dove enough into the code and the complexity of this part to give you an accurate answer. However, I assure you that it is not trivial.

**Comment by thedarkcolour** [comment]
I ran into this same issue while working on a Firebase project that stores documents like:
```json
{
  "subcollection": {
    "uuid1": {
      // data...
    },
    "uuid2": {
      // data...
    },
    "uuid3": {
      // data...
    }
  }
}
```

The "subcollection" field is not used as an attribute for filtering, sorting, or searching.

The reason an object is used is to avoid race conditions that happen when using arrays in Firestore. So, this is not a solution:
```json
{
  "subcollection": [ 
    {
      "id": "uuid1",
      // data...
    }, 
    {
      "id": "uuid2",
      // data...
    }, 
    {
      "id": "uuid3",
      // data...
    }
  ]
}
```

This is a massive pain.
